pytensor/link/jax/dispatch/random.py

```diff
@@ -450,6 +443,42 @@ def sample_fn(rng, size, dtype, n, p):
     return sample_fn
 
 
+def _jax_multinomial(n, p, shape=None, key=None):
+    if jnp.shape(n) != jnp.shape(p)[:-1]:
+        broadcast_shape = jax.lax.broadcast_shapes(jnp.shape(n), jnp.shape(p)[:-1])
+        n = jnp.broadcast_to(n, broadcast_shape)
+        p = jnp.broadcast_to(p, broadcast_shape + jnp.shape(p)[-1:])
+
+    if shape is not None:
+        broadcast_shape = jax.lax.broadcast_shapes(jnp.shape(n), shape)
+        n = jnp.broadcast_to(n, broadcast_shape)
+
+    else:
+        shape = p.shape[:-1]
+
+    p = p / jnp.sum(p, axis=-1, keepdims=True)
+    binom_p = jnp.moveaxis(p, -1, 0)[:-1, ...]
+
+    sampling_rng = jax.random.split(key, binom_p.shape[0])
```

Comment:
It seems you don't have enough rng splits here?, Unless you split it inside the scan as well? You will need a total of `(prod(size) *  binom_p.shape[0]` unique keys if I'm not mistaken.

Every binomial inside the scan needs a unique key, now it's reusing the same for every draw of the multinomial

Reply:
Hmm the scan is taking all the draws simultaneously... In that case your split is probably correct, JAX will further split inside the binomial to draw each draw